source/synthDrivers/_espeak.py

```diff
@@ -123,23 +126,38 @@ def __eq__(self, other):
 @t_espeak_callback
 def callback(wav,numsamples,event):
 	try:
-		global player, isSpeaking, lastIndex
+		global player, isSpeaking, _numBytesPushed
 		if not isSpeaking:
 			return 1
+		indexes = []
 		for e in event:
 			if e.type==espeakEVENT_MARK:
-				lastIndex=int(e.id.name)
+				indexNum = int(e.id.name)
+				# e.audio_position is ms since the start of this utterance.
+				# Convert to bytes since the start of the utterance.
+				# samplesPerSec * 2 bytes per sample / 1000 ms per sec gives us bytes per ms.

```

Comment:
Rather than write this as a comment, the same meaning could be delivered with named constants:
```
BYTES_PER_SAMPLE = 2
MS_PER_SEC = 1000
bytesPerMS = player.samplesPerSec * BYTES_PER_SAMPLE  / MS_PER_SEC
indexByte = e.audio_position * bytesPerMS
```